gecode/flatzinc/flatzinc.cpp

```diff
@@ -1999,6 +2008,190 @@ namespace Gecode { namespace FlatZinc {
+          base += 1;
+        }
+        assert(base == on_restart_iv.size());
+
+        // Reduce on_restart_iv to only the variables required in a solution
+        IntVarArray tmp(*this, restart_data().on_restart_iv_sol);
+        for (int i = 0; i < restart_data().on_restart_iv_sol; ++i) {
+          tmp[i] = on_restart_iv[i];
+        }
+        on_restart_iv = tmp;
+
+        ret = true;
+      }
+      if (on_restart_bv.size() > 0) {
+        int base = 0;
+
+        // Assign the sol_bool values (use lb if no solution is known)
+        const FlatZincSpace* last = static_cast<const FlatZincSpace*>(mi.last());
+        assert(last == nullptr || last->on_restart_bv.size() == restart_data().on_restart_bv_sol);
+        for (int i = 0; i < restart_data().on_restart_bv_sol; ++i) {
+          BoolVar& outVar = on_restart_bv[base + restart_data().on_restart_bv_sol + i];
+          int solVal = (last != nullptr) ? last->on_restart_bv[i].val() : on_restart_bv[base + i].min();
+          rel(*this, outVar, IRT_EQ, solVal);
+        }
+        base += restart_data().on_restart_bv_sol * 2;
+
+        // Assign last_val_bool values
+        for (int i = 0; i < restart_data().last_val_bool.size(); ++i) {
+          rel(*this, on_restart_bv[base + i], IRT_EQ, restart_data().last_val_bool[i]);
+        }
+        base += restart_data().last_val_bool.size();
+        assert(base == on_restart_bv.size());
+
+        // Reduce on_restart_bv to only the variables required in a solution
+        BoolVarArray tmp(*this, restart_data().on_restart_bv_sol);
+        for (int i = 0; i < restart_data().on_restart_bv_sol; ++i) {
+          tmp[i] = on_restart_bv[i];
+        }
+        on_restart_bv = tmp;
+
+        ret = true;
+      }
+#ifdef GECODE_HAS_SET_VARS
+      // TODO: Is there a way to assign SetVar to IntSet without creating a SetVar for the latter?
+      if (on_restart_sv.size() > 0) {
+        int base = 0;
+
+        // Assign the sol_set values (use lb if no solution is known)
+        const FlatZincSpace* last = static_cast<const FlatZincSpace*>(mi.last());
+        assert(last == nullptr || last->on_restart_sv.size() == restart_data().on_restart_sv_sol);
+        for (int i = 0; i < restart_data().on_restart_sv_sol; ++i) {
+          SetVar& outVar = on_restart_sv[base + restart_data().on_restart_sv_sol + i];
+          if (last != nullptr) {
+            auto solVar =last->on_restart_sv[i];
+            rel(*this, outVar, SRT_EQ, solVar);
+          } else {
+            Set::GlbRanges<Set::SetView> lb(on_restart_sv[base + i]);
+            IntSet lbs(lb);
+            SetVar fixed_var(*this, lbs, lbs);
+            rel(*this, outVar, SRT_EQ, fixed_var);
+          }
+        }
+        base += restart_data().on_restart_bv_sol * 2;
+
+        // Assign last_val_set values
+        for (int i = 0; i < restart_data().last_val_set.size(); ++i) {
+          SetVar fixed_var(*this, restart_data().last_val_set[i], restart_data().last_val_set[i]);
```

Comment:
Is there any better way of keeping a set value and then assigning a set variable that value?

Currently I'm keeping `IntSet` type values, and then creating a fixed `SetVar` to use `rel`.

Reply:
There is the dom-constraint, which are constraint usually used for setting domains. See https://www.gecode.org/doc-latest/reference/group__TaskModelSetDom.html#ga8b5263e1bef81c11e4ba9889f0f26568 for relating a single SetVar to an IntSet.

Reply:
Thank you for the pointer! I've amended the code to use `dom` instead